oci/private/download.bzl

```diff
@@ -72,7 +72,8 @@ def _download(
     command = [
         "curl",
         url,
-        "--fail-with-body",
+        "--write-out",
+        "%{http_code}",
```

Comment:
hmm, should this also have something like %{http_body} so we also have the error message written to body?

Reply:
My understanding is that this argument is only what gets written to stdout, but the actual body still gets written according to `--output` below.

Is there something you'd like to get from having the body in stdout?

Reply:
this is different than the behavior of --fail-with-body. we'd like to get the response body written to stderr so we get meaningful errors. 

https://daniel.haxx.se/blog/2021/02/11/curl-fail-with-body/

Reply:
I don't believe that's true, given that we provide the `--output` option. In that case, the response body always gets written to that output file, and nothing is output on stderr, whether you pass `--fail-with-body` or not. 

I tested this based on the example linked there, and got the following output:

```
$ curl --fail-with-body -o output https://example.com/404.html
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1256  100  1256    0     0   1836      0 --:--:-- --:--:-- --:--:--  1849
curl: (22) The requested URL returned error: 404

$ cat output
<!doctype html>
...
```

When we replace this with `--write-out`, the output is the following:
```
curl --write-out "%{http_code}" -o output https://example.com/404.html
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1256  100  1256    0     0    706      0  0:00:01  0:00:01 --:--:--   708
404
$ cat output
<!doctype html>
...
```

The difference here AFAICT is the exit code and the output, that now includes the status code. Does this match your understanding?

Reply:
Oh I see, are you talking about the `curl: (22) The requested URL returned error: 404` error? Sorry, I thought we were talking about the response body.

You are right, that did get suppressed. I'm not sure how to work around that - that just seems to be dropped. We could add the status code to the `fail()`?

Reply:
actually you are right. --output suppresses the --fail-with-body in this case. we should drop that flag.

Reply:
i believe replacing `--fail-with-body` with `--fail` should be equivalent to this PR.

Reply:
Hmm, so that is precisely the difference between the `--fail` and the `--fail-with-body` options. In the case of `--fail`,  then the output file is not logged, and we do not store the response body anywhere.

See the same example:
```
$ curl --fail -o output https://example.com/404.html
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  1256    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404

$ cat output
cat: output: No such file or directory
```


I'm interested in what you prefer here - in my case it was helpful to be able to check the output of the curl command on failures, as you said, it's useful to be able to look at the body, but it also doesn't seem critical. Happy to change to `--fail`, but this PR in its current form preserves the semantics of `--fail-with-body`.